### DISCUSSION TITLE
Create "tracking snippet" to avoid race condition with async script loading

### DISCUSSION THREAD

**Discussion by justusbluemer** [elbwalker/walkerOS#119] [Ideas]
Hi,

if I understand the documentation correctly, one of the options to use walker.js includes calling either
`elb()` (https://docs.elbwalker.com/getting-started/commands) or
`elbwalker.push` (https://docs.elbwalker.com/tagging/user-identification)

This probably works fine when importing walker.js as part of a build process à la `import { elb } from '@elbwalker/walker.js';`, but fails regularly when adding the script to a site with an `async` attribute, as described at the bottom of https://docs.elbwalker.com/getting-started/installation:

`<script async class="elbwalker" src="https://cdn.jsdelivr.net/npm/@elbwalker/walker.js@latest/dist/walker.js" data-default="true"></script>`

When the script is loaded asynchronously but elb() or elbwalker.push() executed immediately afterwards, they might not be defined yet and the call will fail à la `Uncaught ReferenceError: elbwalker is not defined`.

Google Tag Manager and other libraries circumvent this by providing a standardized "tracking snippet" that defines dataLayer (GTM) or fbq (Meta Pixel) directly in the snippet and processes any data pushed through them once the main script is loaded.
Judging from experience, adding a tracking code snippet "manually" to a page template is still the preferred way of loading scripts, even (sometimes especially) with large enterprise companies. Maybe that's an opportunity to make the existing installation instructions even easier.

(btw the CDN snippet at the bottom of [the documentation](https://docs.elbwalker.com/getting-started/installation) doesn't work out of the box; "LATEST" in the URL should be a lowercase "latest")

**Comment by alexanderkirtzel** [comment]
We fixed the documentation and it should be easier to understand now.

I added this note
```js
// Or define the elb function manually in the browser as an alternative
function elb(){(window.elbLayer = window.elbLayer || []).push(arguments);}
```

The walker comes with async support but the docs were wrong.
And using the elbwalker.push is recommended only for internal purposes. Also fixed

Thank you @justusbluemer for the detailed description. Hope it's better now!
